src/commands/domain.rs

```diff
@@ -0,0 +1,106 @@
+use std::time::Duration;
+
+use anyhow::bail;
+use is_terminal::IsTerminal;
+
+use crate::consts::TICK_STRING;
+
+use super::*;
+
+/// Generates a domain for a service if there is not a railway provided domain
+// Checks if the user is linked to a service, if not, it will generate a domain for the default service
+#[derive(Parser)]
+pub struct Args {}
+
+pub async fn command(_args: Args, _json: bool) -> Result<()> {
+    let configs = Configs::new()?;
+
+    let client = GQLClient::new_authorized(&configs)?;
+    let linked_project = configs.get_linked_project().await?;
+
+    let vars = queries::project::Variables {
+        id: linked_project.project.to_owned(),
+    };
+
+    let res = post_graphql::<queries::Project, _>(&client, configs.get_backboard(), vars).await?;
+
+    let body = res.data.context("Failed to retrieve response body")?;
+
+    if body.project.services.edges.is_empty() {
+        bail!("No services found for project");
+    }
+
+    // If there is only one service, it will generate a domain for that service
+    let service = if body.project.services.edges.len() == 1 {
+        body.project.services.edges[0].node.clone().id
+    } else {
```

Comment:
Does line 35 return? If so, we should be using explicit returns

Implicit returns are fine for small functions like map/filter/return but should be avoided, for clarity, here

Also, if the else is going to bail, we should flip the branches and have an early return to remove the else case

Reply:
This another instance of a conditional value